src/main/java/com/eighthlight/SocketWrapper.java

```diff
@@ -0,0 +1,45 @@
+package com.eighthlight;
+import java.net.*;
+import java.io.*;
+
+public class SocketWrapper implements SocketInterface {
+    private ServerSocket serverSocket;
+    private Socket clientSocket;
+    private PrintWriter out;
+    private BufferedReader in;
+
+    public void createAndListen(int port) {
+        try {
+            serverSocket = new ServerSocket(port);
+            clientSocket = serverSocket.accept();
+            out = new PrintWriter(clientSocket.getOutputStream(), true);
+            in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
+            out.println("Enter goodbye to exit");
+        } catch(IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public String receiveData() {
+        try {
+            return in.readLine();
+        } catch(IOException e) {
+            return e.getMessage();
+        }
+    }
+
+    public void sendData(String data) {
+        out.println("Server says:" + data);
```

Comment:
Is this a temporary stub? I would've expected the data to be sent to the client socket (which might also throw an exception).